### PR TITLE
Adding an Icon component, Fine tuning Tab, Changing the structure

### PR DESCRIPTION
**Pull request by Duncid**
All in the title.

**Noticeable change:**

Changed the API of Tab.
name -> label
title -> hideLabel (optional)

### CODE REVIEW

**Review by spolu — APPROVED** [review]
LGTM

But what is an Icon component??

**Comment by spolu** [comment]
I don't quite see why we need it? would love to undertsand :+1:

**Comment by Duncid** [comment]
> I don't quite see why we need it? would love to understand 👍

I might remove indeed.
It's usually helpful to control consistency of icon sizes across the board — the components allows for instance to restrict size or color choices to a given range.

**Comment by spolu** [comment]
Roger that :+1: good to merge :+1:

File: sparkle/src/components/Tab.tsx [hunk 1]
```diff
@@ -63,7 +64,7 @@ export function Tab({ tabs, onTabClick, className = "" }: TabProps) {
         : iconClasses.default;
 
       const finalTabClasses = classNames(
-        "flex gap-x-2 text-sm px-4 py-3 border-b-2 transition-colors duration-200 whitespace-nowrap",
+        "group flex gap-x-2 text-sm font-medium px-4 py-3 border-b-2 transition-colors duration-200 whitespace-nowrap",
```
Comment (spolu) [inline sparkle/src/components/Tab.tsx:67]: based on your Figma it should be pl-3 pr-4 instead of px-4 right? Just checking I get this right :)

Comment (Duncid) [reply sparkle/src/components/Tab.tsx:67]: I'm doing back and forth between figma and code see what works and don't.
I tried to visually align the icon, but it breaks a bucnh of stuff, went back to a more simple px-4

Comment (spolu) [reply sparkle/src/components/Tab.tsx:67]: Roger that :+1:

**Pull request merged**